### ISSUE TITLE
Save memory usage during inference using recent PyTorch versions

### ISSUE THREAD

**Issue by Shrajan** [MIC-DKFZ/nnUNet#2881]
Hi nnUNet team and users,

`Line 138` in `nnunetv2/utilities/label_handling/label_handling.py` contains `logits = logits.float()` which doubles memory consumption during inference (validation and testing) . The original rationale for this conversion was added because "softmax, etc. is not implemented for half", which is true for older PyTorch versions (if I remember correctly, for torch<2.2.0).

#### Suggested Solution
You can either comment out `line 138`, or consider adding a PyTorch version check to conditionally apply this conversion only when necessary:
```
if torch.__version__ < "2.2.0":  # or whatever the minimum version is
    logits = logits.float()
```

So far, I have not observed any accuracy degradation after removing the conversion.

**Comment by vmiller987** [comment]
I will look at this as I have slightly larger than average CT scans. Thanks for sharing, hopefully this helps.

<img width="101" height="66" alt="Image" src="https://github.com/user-attachments/assets/047ecd28-2108-4aee-aecc-42a27733897e" />
